src/app/clusters/operational-credentials/operational-credentials.cpp

```diff
@@ -33,33 +36,193 @@
+CHIP_ERROR writeAdminsIntoFabricsListAttribute(void)
+{
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to writeAdminsIntoFabricsListAttribute");
+
+    // Loop through admins
+    auto pairing = GetGlobalAdminPairingTable().cbegin();
+    int32_t fabricIndex = 0;
+    while (pairing != GetGlobalAdminPairingTable().cend())
+    { 
+        NodeId nodeId = pairing->GetNodeId();
+        uint64_t fabricId = pairing->GetFabricId();
+        uint16_t vendorId = pairing->GetVendorId();
+
+        // Skip over uninitialized admins
+        if (nodeId != kUndefinedNodeId && fabricId != kUndefinedFabricId && vendorId != kUndefinedVendorId)
+        {
+            if (writeFabric(fabricId, nodeId, vendorId, fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write admin with fabricId %" PRIX64 " in fabrics list", fabricId);
+            }
+            fabricIndex ++;
+        }
+        pairing++;
+    }
+
+    // Store the count of fabrics we just stored
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Stored %" PRIX32 " admins in fabrics list attribute.", fabricIndex);
+    if (writeFabricAttribute(0, ZCL_FABRICS_ATTRIBUTE_ID, (uint8_t *) &fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write admin count %" PRIX32 " in fabrics list", fabricIndex);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+AdminPairingInfo * retrieveCurrentAdmin()
+{
+    uint64_t fabricId = emberAfCurrentCommand()->source;
+    uint64_t nodeId;
+    CHIP_ERROR err = ConfigurationMgr().GetDeviceId(nodeId);
+    if (err == CHIP_NO_ERROR)
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Finding admin with fabricId  %" PRIX64 ".", fabricId);
+        return GetGlobalAdminPairingTable().FindAdmin(fabricId, nodeId);
+    }
+    return nullptr;
+}
+
+class OpCredsAdminPairingTableDelegate : public AdminPairingTableDelegate
+{
+
+    void OnAdminDeletedFromStorage(AdminId adminId) override
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Admin %" PRIX16 " was deleted from admin storage.");
+        writeAdminsIntoFabricsListAttribute();
+    }
+    void OnAdminRetrievedFromStorage(AdminId adminId, FabricId fabricId, NodeId nodeId) override
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Admin %" PRIX16 " was retrieved from storage. FabricId %" PRIX64
+                                            ", NodeId %" PRIX64 ", VendorId %" PRIX64, adminId, fabricId, nodeId);
+        writeAdminsIntoFabricsListAttribute();
+    }
+    void OnAdminPersistedToStorage(AdminId adminId, FabricId fabricId, NodeId nodeId) override
+    {
+        emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Admin %" PRIX16 " was persisted to storage. FabricId %" PRIX64
+                                            ", NodeId %" PRIX64 ", VendorId %" PRIX64, adminId, fabricId, nodeId);
+        writeAdminsIntoFabricsListAttribute();
```

Comment:
Why do we need to write when an admin is persisted _and_ when one is retrieved?  Might be worth documenting this, at least.

Reply:
Will document, but one is retrieved as when the server starts, we load all persisted admins into the cached admin pairing table so then I load it in the attributes. When a new commissioner pairs with accessory, a new admin is created, added to the cached admin pairing table and persisted so we want to add it to the attributes as well.

Reply:
Makes sense.  Just document, please.